### PR TITLE
feat(lambda-nodejs): add sourcesContent in BundlingOptions

### LINKED ISSUE #17256: @aws-cdk/aws-lambda-nodejs: Esbuild bundling option to disable sourcesContent
### Description

With the `NodeJsFunction` construct, We have observed that the source code is included as part of the source maps. For production builds, We do not need this as we are not going to debug the lambda. 

### Use Case

For production lambdas, it is desirable to omit the source code in source maps generated by esbuild through `NodeJsFunction` construct

### Proposed Solution

In `esbuild`, there is an option we can set to disable this to not to include the source code as per `esbuild` docs -https://esbuild.github.io/api/#sources-content 

We can add a property (sourcesContent) to `BundlingOptions` type and allow the function to enable/disable the inclusion of sourcesContent into the source map generated.

### Other information

_No response_

### Acknowledge

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### PR DESCRIPTION
**Pull request by suds-sky**
Support excluding sourcesContent from `esbuild` generated source map 

Closes #17256

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by suds-sky** [comment]
@nija-at Can you please check this. Tx.

File: packages/@aws-cdk/aws-lambda-nodejs/README.md [hunk 1]
```diff
@@ -174,6 +174,7 @@ new lambda.NodejsFunction(this, 'my-handler', {
     minify: true, // minify code, defaults to false
     sourceMap: true, // include source map, defaults to false
     sourceMapMode: SourceMapMode.INLINE, // defaults to SourceMapMode.DEFAULT
+    sourcesContent: false, // include original source into source map, defaults to true
```
Comment (jogold) [inline packages/@aws-cdk/aws-lambda-nodejs/README.md:177]:
```suggestion
    sourcesContent: false, // do not include original source into source map, defaults to true
```

**Comment by jogold** [comment]
@nija-at LGTM

**Comment by suds-sky** [comment]
Thanks @jogold taking a look at this PR. Is there anything else I have to do from my side to get this merged and released.

**Comment by jogold** [comment]
> Is there anything else I have to do from my side to get this merged and released.

No, the PR is fine and should be approved when someone from the CDK team has time to have a look at it.

**Review by peterwoodworth — APPROVED** [review]
Looks great, thanks for the contribution!

**Pull request merged**